internal/pkg/artifact/config.go

```diff
@@ -74,17 +111,19 @@ func (r *Reloader) Reload(rawConfig *config.Config) error {
 	}
 	cfg := &reloadConfig{}
 	if err := rawConfig.Unpack(&cfg); err != nil {
-		return errors.New(err, "failed to unpack config during reload")
+		return err
 	}
 
 	var newSourceURI string
-	if cfg.FleetSourceURI != "" {
+	if fleetURI := strings.TrimSpace(cfg.FleetSourceURI); fleetURI != "" {
 		// fleet configuration takes precedence
-		newSourceURI = cfg.FleetSourceURI
-	} else if cfg.SourceURI != "" {
-		newSourceURI = cfg.SourceURI
+		newSourceURI = fleetURI
+	} else if sourceURI := strings.TrimSpace(cfg.SourceURI); sourceURI != "" {
+		newSourceURI = sourceURI
 	}
 
+	newSourceURI = strings.TrimSpace(newSourceURI)
```

Comment:
Question?

Are we validating anywhere that the source URI is valid? By valid I mean the format is valid and not reaching the host?

Reply:
no we dont do that download will fail when trying to use something not usable